test/node-rpc-test.js

```diff
@@ -177,6 +177,13 @@ describe('RPC', function() {
       const connectionsCnt = await nclient.execute('getconnectioncount', []);
       assert.strictEqual(connectionsCnt, 1);
     });
+
+    it('should rpc setban a peer',async () => {
+      const result = await nclient.execute('setban',[`127.0.0.1:${ports.p2p}`,'add']);
+      assert.strictEqual(result,null);
+      const listbanned = await nclient.execute('listbanned',[]);
+      assert.strictEqual(listbanned.length, 1);
```

Comment:
what you've asserted here is that a hostname has been added to the ban list (although since we didn't assert an empty list at the beginning of the test, we don't really know what happened). I think you should also assert that the peer that WAS connected at the beginning of the test is now gone. You can add more tests with `getconnectioncount` and `getpeerinfo`.

I added these checks myself locally and discovered that you were not actually banning a connected peer. That's how I was able to debug and noticed the port was wrong. TEST THE TEST! ;-)